wikitexthtml/render/preprocess.py

```diff
@@ -73,6 +85,44 @@ def hr_line(instance: WikiTextHtml, body: str) -> str:
     return regex.sub(r"\n----+\n", r"\n\n<hr />\n", body)
 
 
+def code_block(instance: WikiTextHtml, body: str) -> str:
+    # Use Pygments to generate highlighted code blocks
+
+    # The SYNTAXHIGHLIGHT_PATTERN Generates 6 groups:
+    #   The first group (match[0]) is the entire match
+    #   The third group (match[2]) is the language
+    #   The fourth group (match[3]) is a boolean to specify line numbering
+    #   The fifth group (match[4]) is the line number to start at
+    #   The sixth group (match[5]) is the code to highlight
+    # You can also reference the groups by name, i.e. match.group("lang")
+    # An interactive example with test strings is at https://regex101.com/r/9KjJYa/1
+
+    for match in reversed(list(SYNTAXHIGHLIGHT_PATTERN.finditer(body))):
+        lang = match.group("lang") if match.group("lang") else "unknown"
+        lineBoolean = True if match.group("line") else False
+        lineStart = int(match.group("lineStart")) if match.group("lineStart") else 1
+        content = match.group("code")
+
+        try:
+            # If language was provided, use that lexer
```

Comment:
Sentences should end with dots. Goes for all your comments :)

Reply:
You will be very dissapointed with the comment conventions currently in use by an API used by a global company with 4,000+ locations, then ;)

(I'll clean these up)